### PR TITLE
feat(98): add Product card tests and stories, refactoring main component

### PR DESCRIPTION
**Pull request by premiumderyn**
This PR refactors the ProductCard component, and introduces stories and unit tests to ensure UI reliability.
* Adjust display of elements as in the design
* Created `ProductCard.stories.tsx` to document various edge cases.
* Added unit tests in `ProductCard.test.tsx`  to cover rendering and hover state visibility.
image_2026-03-05_17-02-00

### CODE REVIEW

**Review by yunaree — APPROVED** [review]

File: src/components/ProductCard/ProductCard.tsx [hunk 1]
```diff
@@ -6,28 +6,48 @@ interface ProductCardProps {
...
+        <div className="absolute right-0 bottom-0 left-0 translate-y-full px-4 pb-4 opacity-0 transition-all duration-300 group-hover:translate-y-0 group-hover:opacity-100">
+          <Button className="w-full rounded-md bg-[#141718] py-3 text-sm font-medium text-white transition-all outline-none hover:border hover:border-white">
+            Add to Cart
+          </Button>
+        </div>
+        <button
+          aria-label="Add to wishlist"
+          className="absolute top-3 right-3 flex h-8 w-8 items-center justify-center rounded-full bg-white opacity-0 shadow-md transition-all duration-300 group-hover:opacity-100 hover:scale-110"
+        >
+          <svg
```
Comment (Fronik123) [inline src/components/ProductCard/ProductCard.tsx]: you can use HeartIcon from lucide-react

**Review by Fronik123 — APPROVED** [review]

**Review by dianajnxv — APPROVED** [review]

File: src/components/ProductCard/ProductCard.tsx (hunk 1 above)
Comment (premiumderyn) [reply src/components/ProductCard/ProductCard.tsx]: Fixed
